functions-iterators/starter.js

```diff
@@ -29,52 +29,214 @@ This will help us track your progress and it will give you practice writing good
 
 // PART 0: Just DO it!!
 
+function justDoIt(name){
 
+  console.log(`${name} JUST DO IT!`);
+}
+
+justDoIt("Khalid");
 
 // PART 1: Big or Small String?
 
+function bigOrSmallString(string){
+
+  if(string.length > 10){
+    console.log("This word is loooooong!");
+
+  }else{
+    console.log("This word is short 😬");
+  }
+}
 
+bigOrSmallString("Sunday fun day ^_^");
 
 // PART 2: Odd or Even String Length?
 
+function oddOrEvenString(string){
+
+  if(string.length % 2 ===0){
+
+    console.log("This string length is TOTALLY even!");
+  }else if(string.length % 2 > 0 ){
+
+    console.log("This string's length is odd..");
+  }
+}
 
+oddOrEvenString("General Assembly");
 
 // PART 3: Median
+var myArray = [221,46,42,988,9]
+
+function medianOfArray (){
+
+
+ var sortedArray = myArray.sort(function(a,b){
+
+  return a - b;
+
+
+});
+
+var theMiddle = Math.floor(sortedArray.length  / 2);
+
+console.log(sortedArray);
+
+
+  if(sortedArray.length % 2 === 0) {
+    console.log("The Median is " +(sortedArray[theMiddle - 1]  + sortedArray[theMiddle]));
+  } else {
+    console.log("The Median is "+ sortedArray[theMiddle]);
+}
+}
+
 
 
 
+
+medianOfArray();
+
+
 // PART 4: Sum Array
 
+var sumArray = [44, 22, 33].reduce(sum, 0);
 
+function sum(a, b) {
+    return a + b;
+}
+
+console.log(sumArray);
 
 // PART 5: Vowel Count
 
+function vowelCount(str) {
+  var matches = str.match(/[aeiou]/gi);
+
+  if(matches.includes("i") ||matches.includes("o") || matches.includes("a") || matches.includes("e") || matches.includes("u") ){
+
+    return matches.length;
+  }else{
+
+    return 0;
+  }
+
+}
+
+console.log(vowelCount("Honey Badger"));
+
 
 
 // PART 6: Initials
 
+function initials(personName){
 
+  var ini = personName.match(/\b(\w)/g);
+
+  var str1 = ini.join();
+  console.log(ini);
```

Comment:
great way to use regular expression just `console.log(str1)` to see the final result